fossology/fossology
# feat(spdx): update reports to v-2.3

## Description

Update SPDX RDF and tag:value reports to specification version 2.3.

### Changes

1. General:
    1. Use organization name from "Report Header Text" filed from customize page.
    2. Use tool version as `fossology-<version>` where version is build version (e.g. `4.2.1.19`)
    3. Fix report creation timestamp to use UTC timezone.
    4. Fix verification checksum by excluding parent file from calculation (parent file = uploaded file).
    5. Include obligations and acknowledgements in reports.
    6. Update license list version to 3.19
2. RDF files:
    1. Change extension from `-spdx.rdf` to `.spdx.rdf`
    2. Use `xml:base` to reduce URI lengths.
    3. If concluded licenses include `Dual-license`, then use `DisjunctiveLicenseSet` (to denote `OR`), otherwise use `ConjunctiveLicenseSet` (to denote `AND`).
    4. If there is only one concluded license, do not use any set.
    5. Print license acknowledgements in `<spdx:File>` using `<spdx:attributionText>`.
    6. Use general assessment as `<rdfs:comment>` in `<spdx:Package>`.
    7. Use license obligations in `<spdx:Package>` using `<spdx:attributionText>`.
3. Tag-value files:
    1. Print license acknowledgements using `FileAttributionText: <text> ... </text>`.
    2. Print `PackageVersion:` and `ReleaseDate:` from report conf.
    3. Print general assessment as `PackageComment: <text> ... </text>`.
    4. Print license obligations using `PackageAttributionText: <text> ... </text>`.

## How to test

1. Generate SPDX reports in RDF and tag value format.
2. Check if the reports looks good.
3. Validate reports using SPDX tools or online validator at: https://tools.spdx.org/app/validate/
4. Check if other reports are not broken (DEP5 and SPDX CSV).
5. Check for issues reported in #1894

Depends on #2363

Closes #1968
Closes #1894

<a href="https://gitpod.io/#https://github.com/fossology/fossology/pull/2368"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
